src/token/PendingAirdropId.js

```diff
@@ -11,18 +31,27 @@ export default class PendingAirdropId {
     /**
      *
      * @param {object} props
-     * @param {AccountId} props.senderId
-     * @param {AccountId} props.receiverId
+     * @param {AccountId} [props.senderId]
+     * @param {AccountId} [props.receiverId]
      * @param {TokenId?} props.tokenId
      * @param {NftId?} props.nftId
      */
     constructor(props) {
-        this.senderId = props.senderId;
-        this.receiverId = props.receiverId;
+        this._senderId = null;
+        this._receiverId = null;
+        this._tokenId = null;
+        this._nftId = null;
+
+        if (props.receiverId) {
```

Comment:
IMO use setters or direct variable assignments for all properties here to maintain consistency.